tests/subscriptions/test_scheduler_processing_strategy.py

```diff
@@ -418,3 +438,130 @@ def test_tick_buffer_with_commit_strategy() -> None:
         mock.call(make_message_for_next_step(message_0_0, True)),
         mock.call(make_message_for_next_step(message_1_1, False)),
     ]
+
+
+def test_scheduled_subscription_queue() -> None:
+    queue = ScheduledSubscriptionQueue()
+    assert len(queue) == 0
+    assert queue.peek() is None
+    with pytest.raises(IndexError):
+        queue.popleft()
+
+    epoch = datetime(1970, 1, 1)
+    partition = Partition(Topic("test"), 0)
+
+    tick_message = Message(
+        partition,
+        1,
+        CommittableTick(
+            Tick(
+                0,
+                offsets=Interval(1, 3),
+                timestamps=Interval(epoch, epoch + timedelta(minutes=2)),
+            ),
+            True,
+        ),
+        epoch,
+        2,
+    )
+
+    future: Future[Message[KafkaPayload]] = Future()
+
+    queue.append(tick_message, deque([future]))
+
+    assert len(queue) == 1
+    assert queue.peek() == TickSubscription(tick_message, future, should_commit=True)
+    assert queue.popleft() == TickSubscription(tick_message, future, should_commit=True)
+    assert len(queue) == 0
```

Comment:
Please test the `popleft` method with more than one tick and with more than one future per tick. That is the interesting part of this class.